swift-codec/src/main/java/com/facebook/swift/codec/metadata/ReflectionHelper.java

```diff
@@ -238,67 +242,52 @@ public static Method findAnnotatedMethod(Class<?> configClass, Class<? extends A
+            new CacheLoader<Executable, List<String>>()
+            {
+                @Override
+                public List<String> load(Executable executable)
+                {
+                    return getParameterNames(executable);
+                }
+            });
 
-    public static String[] extractParameterNames(AccessibleObject methodOrConstructor)
+    public static List<String> extractParameterNames(Executable methodOrConstructor)
     {
-        String[] names = PARANAMER.lookupParameterNames(methodOrConstructor);
-        return names;
+        return PARAMETER_NAMES.getUnchecked(methodOrConstructor);
     }
 
-    private static class ThriftFieldParanamer extends AnnotationParanamer
+    private static List<String> getParameterNames(Executable executable)
     {
-        @Override
-        protected String getNamedValue(Annotation annotation)
-        {
-            if (annotation instanceof ThriftField) {
-                String name = ((ThriftField) annotation).name();
-                if (!name.isEmpty()) {
-                    return name;
-                }
-            }
-            return super.getNamedValue(annotation);
-        }
+        requireNonNull(executable, "executable is null");
 
-        @Override
-        protected boolean isNamed(Annotation annotation)
-        {
-            return annotation instanceof ThriftField || super.isNamed(annotation);
+        if (executable.getParameterCount() == 0) {
+            return emptyList();
         }
-    }
 
-    private static class GeneralParanamer implements Paranamer
-    {
-        @Override
-        public String[] lookupParameterNames(AccessibleObject methodOrConstructor)
-        {
-            String[] names;
-            if (methodOrConstructor instanceof Method) {
-                Method method = (Method) methodOrConstructor;
-                names = new String[method.getParameterTypes().length];
-            }
-            else if (methodOrConstructor instanceof Constructor<?>) {
-                Constructor<?> constructor = (Constructor<?>) methodOrConstructor;
-                names = new String[constructor.getParameterTypes().length];
-            }
-            else {
-                throw new IllegalArgumentException("methodOrConstructor is not an instance of Method or Constructor but is " + methodOrConstructor.getClass().getName());
-            }
-            for (int i = 0; i < names.length; i++) {
-                names[i] = "arg" + i;
-            }
-            return names;
+        // first try to get the parameter names from the ThriftField annotations
+        List<Optional<String>> parameterNamesFromThriftField = Arrays.stream(executable.getParameters())
+                .map(ReflectionHelper::getThriftFieldParameterName)
+                .collect(toImmutableList());
+        if (parameterNamesFromThriftField.stream().allMatch(Optional::isPresent)) {
```

Comment:
I don't see a need for this special case. It seems the below loop will give the same behavior.